bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/DiscoveryResultImpl.java

```diff
@@ -43,7 +43,7 @@ public class DiscoveryResultImpl implements DiscoveryResult {
     private @Nullable String representationProperty;
     private @NonNullByDefault({}) DiscoveryResultFlag flag;
     private @NonNullByDefault({}) String label;
-    private long timestamp;
+    private Instant timestamp = Instant.MIN;
```

Comment:
@J-N-K - unfortunately I ran into this nasty regression today:
```
2024-01-08 21:43:22.754 [ERROR] [re.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.openhab.core.storage.json.internal.StorageEntry@3df4a8fa'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was NUMBER at path $.timestamp
```

so now realize that this object is deserialized from the JSON file `userdata/org.openhab.core.config.discovery.DiscoveryResult.json`.

The easy solution is to revert this variable back to `long`. Alternately we can keep `Instant`, but implement custom deserialization that can handle both. WDYT? Right at this moment I'm not sure where this is deserialized.